File: UsingProcessLifecycleOwnerForLibs/app/src/main/java/github/nisrulz/sample/usingprocesslifecycleownerforlibs/MyApplication.java
package github.nisrulz.sample.usingprocesslifecycleownerforlibs;

import android.app.Application;
import androidx.lifecycle.ProcessLifecycleOwner;
import github.nisrulz.sample.lib.MyLib;


public class MyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        // setup process lifecycle tracking
        MyLib myLib = new MyLib(ProcessLifecycleOwner.get().getLifecycle());

    }
}

File: UsingProcessLifecycleOwnerForLibs/app/src/main/java/github/nisrulz/example/usingprocesslifecycleownerforlibs/MyApplication.kt
package github.nisrulz.example.usingprocesslifecycleownerforlibs

import android.app.Application
import androidx.lifecycle.ProcessLifecycleOwner
import github.nisrulz.android_lib.MyLib

class MyApplication : Application() {
    lateinit var mylib: MyLib

    override fun onCreate() {
        super.onCreate()

        // setup process lifecycle tracking
        mylib = MyLib(ProcessLifecycleOwner.get().lifecycle)
    }
}